packages/components/button/src/button.js

```diff
@@ -28,7 +29,16 @@ export class TSButton extends TSElement {
 			<button ?disabled="${this.disabled}">
 				${this.icon
 					? html`
-							<ts-icon icon="${this.icon}" size="medium"></ts-icon>
+							<ts-icon
+								icon="${this.icon}"
+								size="medium"
+								type="${this.type === types.PRIMARY ||
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3AS3358)